pyodk/_utils/session.py

```diff
@@ -115,7 +115,7 @@ def response_or_error(
             response.raise_for_status()
         except HTTPError as e:
             msg = (
-                f"The request to {url} failed."
+                f"The request to {self.base_url}{url} failed."
```

Comment:
self.urljoin(url) will add the base_url and handle stripping slash. Better to use same function.